src/pages/home/ReportScreen.js

```diff
@@ -160,12 +156,29 @@ function ReportScreen({
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const {isSmallScreenWidth} = useWindowDimensions();
-
-    const firstRenderRef = useRef(true);
+    const {isOffline} = useNetwork();
     const flatListRef = useRef();
     const reactionListRef = useRef();
+    const firstRenderRef = useRef(true);
+    const reportIDFromRoute = getReportID(route);
+    const reportActionIDFromRoute = lodashGet(route, 'params.reportActionID', null);
+    const shouldTriggerLoadingRef = useRef(!!reportActionIDFromRoute);
     const prevReport = usePrevious(report);
     const prevUserLeavingStatus = usePrevious(userLeavingStatus);
+    const [isPrepareLinkingToMessage, setLinkingToMessage] = useState(!!reportActionIDFromRoute);
+
+    const reportActions = useMemo(() => {
+        if (!!allReportActions && allReportActions.length === 0) {
+            return [];
+        }
+        const sortedReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(allReportActions, false);
+        const currentRangeOfReportActions = ReportActionsUtils.getContinuousReportActionChain(sortedReportActions, reportActionIDFromRoute);
+        // eslint-disable-next-line rulesdir/prefer-underscore-method
+        const reportActionsWithoutDeleted = currentRangeOfReportActions.filter((item) => ReportActionsUtils.shouldReportActionBeVisible(item, item.reportActionID));
```

Comment:
Let's rename this variable to be a bit simpler and not suppress lint:

```suggestion
        const visibleReportActions = _.filter(currentRangeOfReportActions, (reportAction) => ReportActionsUtils.shouldReportActionBeVisible(reportAction, reportAction.reportActionID));
```

Reply:
Or you could just get rid of the redundant local variable and return:

```js
return _.filter(currentRangeOfReportActions, (reportAction) => ReportActionsUtils.shouldReportActionBeVisible(reportAction, reportAction.reportActionID))
```

Reply:
👍🏻